views/punchIn.js

```diff
@@ -97,7 +102,11 @@ const sendErrorNotifcation = () => {
     position: "right top",
   });
 };
-
+if (localStorage.getItem("passKey") !== "wework") {
```

Comment:
this check should happen on server side not FE. 
Anybody can see passkey is they check the JS on the browser.